### DISCUSSION TITLE
Why does enforceActions only warn instead of throw?

### DISCUSSION THREAD

**Discussion by smikula** [mobxjs/mobx#3912] [Q&A]
I only recently discovered that this changed back in 2020 with [this commit](https://github.com/mobxjs/mobx/commit/f60c913dc5b887bd647cfe41d29abff6b9971d91#diff-56b9c407b0f1b6baa90f4f1d852db42b3915fe79119c9395d31bf8f1fe831b41) from @mweststrate.  Previously when strict mode was enabled, trying to change state outside of an action would throw.  Now it just logs a warning to the console, a warning that — at least my organization — is sure to go ignored.

What was the rationale behind this change?  Would you consider adding an "extra strict" option to go back to the throwing behavior?  At the moment this leaves a pretty big gap for poor code to slip into my codebase.

**Comment by urugator** [comment]
See https://github.com/mobxjs/mobx/issues/3618#issuecomment-1410268462 and the comments below

**Reply by smikula** [reply]
Cool, that gives some context.  I guess the official answer is "patch `console.warn` in your own codebase".
